app/queries/user_behavior_query.rb

```diff
@@ -0,0 +1,86 @@
+class UserBehaviorQuery < BaseQuery
+
+  DATE_STRING_FORMAT = '%d-%m-%Y'.freeze
+
+  def initialize
+    @relation = Person.all.unscoped
+  end
+
+  def call
+    @relation.
+      joins(joins_sql).
+      select(selected_columns).
+      order('people.id, full_name')
+  end
+
+  def data
+    call.map do |rec|
+      { id: rec.id,
+        full_name: rec.full_name,
+        address: rec.address, # can't use location as the name as this is a method on person model
+        team_name: rec.team_name,
+        ancestors: rec.ancestors&.join(' > '),
+        team_role: rec.team_role,
+        login_count: rec.login_count,
+        last_login_at: rec.last_login_at&.strftime(DATE_STRING_FORMAT),
+        updates_count: rec.updates_count
+      }
+    end
+  end
+
+  private
+
+  def joins_sql
+    <<~SQL
+    LEFT JOIN memberships ON memberships.person_id = people.id
+    LEFT JOIN groups ON groups.id = memberships.group_id
+    SQL
+  end
+
+  # rubocop:disable MethodLength
+  def select_ancestors
```

Comment:
Personal preference but, if you're going to disable MethodLength anyway, I'd stick this into a single massive SQL block.

It'll be easier to parse than jumping up and down to different methods.  YMMV